tests/test_isrFunctions.py

```diff
@@ -0,0 +1,402 @@
+        r"""Image transpose.
+
+        Expect height and width to be exchanged.
+        """
+        transposed = ipIsr.transposeMaskedImage(self.mi)
+        self.assertEqual(transposed.getImage().getBBox().getHeight(),
+                         self.mi.getImage().getBBox().getWidth())
+        self.assertEqual(transposed.getImage().getBBox().getWidth(),
+                         self.mi.getImage().getBBox().getHeight())
+
+    def test_interpolateDefectList(self):
+        r"""Defect interpolation.
+
+        Expect number of interpolated pixels to be non-zero.
+        """
+        defectList = isrMock.DefectMock().run()
+        assert len(defectList) > 0
+
+        for fallbackValue in (None, -999.0):
+            for haveMask in (True, False):
+                with self.subTest(fallbackValue=fallbackValue, haveMask=haveMask):
+                    if haveMask is False:
+                        if 'INTRP' in self.mi.getMask().getMaskPlaneDict():
+                            self.mi.getMask().removeAndClearMaskPlane('INTRP')
+                    else:
+                        if 'INTRP' not in self.mi.getMask().getMaskPlaneDict():
+                            self.mi.getMask().addMaskPlane('INTRP')
+                    numBit = maskCountPixels(self.mi, "INTRP")
+                    self.assertGreaterEqual(numBit, 0)
+
+    def test_transposeDefectList(self):
+        r"""Transpose defect list.
+
+        Expect bbox dimension values to flip.
+        """
+        defectList = isrMock.DefectMock().run()
+
+        transposed = ipIsr.transposeDefectList(defectList)
+
+        for d, t in zip(defectList, transposed):
+            assert d.getBBox().getDimensions().getX() == t.getBBox().getDimensions().getY()
+            assert d.getBBox().getDimensions().getY() == t.getBBox().getDimensions().getX()
+
+    def test_makeThresholdMask(self):
+        r"""Thresholding.
+
+        Expect list of defects to have elements.
+        """
+        defectList = ipIsr.makeThresholdMask(self.mi, 200, growFootprints=2, maskName='SAT')
+
+        self.assertGreater(len(defectList), 0)
+
+    def test_interpolateFromMask(self):
+        r"""Interpolate mask.
+
+        Expect number of interpolated pixels to be non-zero.
+        """
+        ipIsr.makeThresholdMask(self.mi, 200, growFootprints=2, maskName='SAT')
+        for growFootprints in (0, 2):
+            with self.subTest(growFootprints=growFootprints):
+                interpMaskedImage = ipIsr.interpolateFromMask(self.mi, 2.0,
+                                                              growFootprints=growFootprints, maskName='SAT')
+                numBit = maskCountPixels(interpMaskedImage, "INTRP")
+                self.assertGreaterEqual(numBit, 0)
+
+    def test_saturationCorrection(self):
+        r"""Saturation correction.
+
+        Expect number of mask pixels with SAT marked to be non-zero.
+        """
+        for interpolate in (True, False):
```

Comment:
Tests of exactly two case are often clearer as two separate tests: `test_saturationCorrectionInterpolate`, `test_saturationCorrectionNoInterpolate`.